app/Http/Controllers/ScoresController.php

```diff
@@ -42,6 +44,11 @@ private static function parseIdOrFail(string $id): int
 
     public function download($rulesetOrSoloId, $id = null)
     {
+        $currentUser = \Auth::user();
+        if(is_api_request() && $currentUser === null) {
```

Comment:
"if api request and guest user"?

Reply:
Didn't you just tell me to use the other method? (is_api_request instead of from_app_url)

Reply:
yes but is it "if api request" or "if not api (aka web) request"?

Reply:
Or do you mean I should negate the is_api_request check